GID Flag and Node.Self Structure

andrewnguyen22 opened the pull request:
Added flag: --gid "myGID"
Added node.Self singleton structure for easy self node properties like GID ^
Closes #104 
Closes #123

Commit:
added self node singleton to access self node

Commit:
fixed constructors for entering and exiting network for node.Self

Commit:
added hosted chains test WIP

iajrz reviewed (changes requested):
[rpc.RunAPIEndpoints](https://github.com/pokt-network/pocket-core/pull/117/files#diff-1318addeee65b25b7124bace036f67d9R63) seems not to be available in this change set. Maybe there was a mixup when staging. This branch can't be built. 
[`ExitGracefully`](https://github.com/pokt-network/pocket-core/pull/117/files#diff-8a1278a5855334feb584493def4bfec5R8) should log the cause for exiting, not just print it.
I expected, because of the issue, that the default GID would be "dummyGID". 0 works too; if this was ignored on purpose, all is well.

Commit:
fixed rpc.RunAPI and logged when gracefully exiting

andrewnguyen22 commented:
Ahh yes, the manual merging is what did me in! Good catch, I fixed and made the changes you requested!

iajrz committed:
Eliminates import cycle between logs and util

this was achieved by moving the `Caller` method to the log package,
it being the only need of the log package to use the `util` package.
Furthermore, because the "caller" method is only used for logging,
it was made private.

iajrz commented:
9e37f20 created an import cycle which prevented the project from building. I eliminated it in d125d79.
